docs/source/integrations/mobile/ios-fairplay-sdk.md

```diff
@@ -87,125 +86,115 @@ More information about ATS can be found in Apple’s TechNote:
+	
+The ContentKeyManager class should look like this:
+
+	import AVFoundation
+ 	import vudrmFairPlay
+ 	
+ 	class ContentKeyManager {
+    
+    /// MARK: Types.
+    
+    /// The singleton for `ContentKeyManager`
+    static let shared: ContentKeyManager = ContentKeyManager()
+    // MARK: Properties.
+    
+    /**
+     The instance of `AssetResourceLoaderDelegate` which conforms to `AVAssetResourceLoaderDelegate` and is used to respond to content key requests
+     from `AVAssetResourceLoader`.
+     */
+    let assetResourceLoaderDelegate: vudrmFairPlay
+    
+    /// The DispatchQueue to use for delegate callbacks.
+    let assetResourceLoaderDelegateQueue = DispatchQueue(label: "com.vualto.vurdrm.fairplay.AssetResourceLoaderDelegateQueue")
+    
+    var drm: vudrmFairPlay?
+    
+	/// MARK: Initialization.
+    
+    private init() {
+        var token = ""
+        var contentID = ""
+        for stream in StreamListManager.shared.streams {
+            if stream.vudrmToken != nil {
+                token = stream.vudrmToken!
+                contentID = stream.name
+                let asset = AVURLAsset(url: URL(string: stream.playlistURL)!)
+                self.drm = vudrmFairPlay(asset: asset, contentID: contentID, token: token)!
+            }
+        }
+        assetResourceLoaderDelegate = self.drm!
+    }
+    
+    func updateResourceLoaderDelegate(forAsset asset: AVURLAsset) {
+        asset.resourceLoader.setDelegate(self.drm, queue: assetResourceLoaderDelegateQueue)
+        }
+    }
+     
+     
 ##### Online Playback / Download and Offline Playback:
 
-This method will request a license for the content from the license server based on the type of token presented for each instance of VUDRM FairPlay. The tokens may be FairPlay Rental or FairPlay Persist. FairPlay Rental token license requests will be provided a streaming content key, which may be used for online streaming only. FairPlay Persist token license requests will provide a persistent content key, which may be used for both online streaming, and offline playback of downloaded content.
+This will request a license for the content from the license server based on the type of token presented for each instance of VUDRM FairPlay. The tokens may be FairPlay Rental or FairPlay Persist. FairPlay Rental token license requests will be provided a streaming content key, which may be used for online streaming only. FairPlay Persist token license requests will provide a persistent content key, which may be used for both online streaming, and offline playback of downloaded content.
```

Comment:
```suggestion
This will request a license for the content from the license server based on the type of token presented for each instance of VUDRMFairPlay. The tokens may be FairPlay Rental or FairPlay Persist. FairPlay Rental token license requests will be provided a streaming content key, which may be used for online streaming only. FairPlay Persist token license requests will provide a persistent content key, which may be used for both online streaming, and offline playback of downloaded content.
```